### PR TITLE
feat: add identity knowledge methods

### PR DESCRIPTION
**Pull request by cowan-macady**
Closing ENG-1850
remove some old tests

### CODE REVIEW

**Review by maaland — CHANGES_REQUESTED** [review]
Make sure you also add `deprecated` wording to the `get_digital_twin()`, `get_digital_twin_by_token()` and `get_digital_twin_by_property()` in the Identity package. As was done in the [Go implementation](https://github.com/indykite/indykite-sdk-go/pull/143/files#diff-c5abb4600a8ce207aba086c29a2fbb15b21057f66fb872be4bdd7c9f35e695a5)

File: indykite_sdk/knowledge/identity_knowledge.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+
+def read(self, path, conditions, input_params={}):
+    """
+    read ingested data
+    :param self:
+    :param path: string
+    :param conditions: string
+    :param input_params: map{string, InputParam}
+    :return: record_error
```
Comment (maaland) [inline indykite_sdk/knowledge/identity_knowledge.py:16]: i don't thin we should call this a *record*_error

**Review by maaland — APPROVED** [review]
Looks fine to me, to the extent I can review Python SDK stuff 😅

File: indykite_sdk/knowledge/identity_knowledge.py (hunk 1 above)
Comment (cowan-macady) [reply indykite_sdk/knowledge/identity_knowledge.py]: sorry, not the response ndeed

File: indykite_sdk/knowledge/identity_knowledge.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+        params = {"id": id}
+        identity_knowledge_response = self.stub.IdentityKnowledge(
+            pb2.IdentityKnowledgeRequest(
+                    operation=Operation.OPERATION_READ.value,
+                    path=path,
+                    conditions=conditions,
+                    input_params=request_input_params(params),
+            )
+        )
+        if not identity_knowledge_response or len(identity_knowledge_response.paths) == 0:
```
[the remark this answers is not in the document]
Comment (cowan-macady) [reply indykite_sdk/knowledge/identity_knowledge.py:59]: In this case, it is to have the same behavior as in go sdk.
Usually, not response means empty response and the value of the response is "None".
In fact a specific handle of the not response is not really necessary. May be I should change that.
If you have some examples about a None which could be an error, please tell me.
Every error is logged, even the uncaught ones with sys.excepthook = logger.handle_excepthook.
May be I should make a complete check of all methods returns in the sdk to see if something is missing.

[the remark this answers is not in the document]
Comment (Septimus4) [reply indykite_sdk/knowledge/identity_knowledge.py:59]: No it seems okay with me, just wanted to make sure

**Review by Septimus4 — APPROVED** [review]

**Review by maaland — APPROVED** [review]

**Pull request merged**